### PR TITLE
fix(General): Generic crash fixes with nil pointer checks

### PR DESCRIPTION
**Pull request by seswarrajan**
**Purpose of PR?**:
1. Added null pointer check to fix crash
2. Added logic for relay to connect only if stream connection interrupts

Fixes #

**Does this PR introduce a breaking change?** No

**If the changes in this PR are manually verified, list down the scenarios covered:**:
For system and nw policy 

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: src/plugin/kubearmor.go [hunk 1]
```diff
@@ -407,7 +404,9 @@ func StartKubeArmorRelay(StopChan chan struct{}, cfg types.ConfigKubeArmorRelay)
 		defer func() {
 			log.Info().Msg("watchlogs returning")
 			KubeArmorRelayStarted = false
-			_ = conn.Close()
+			if conn != nil {
+				_ = conn.Close()
+			}
```
Comment (Ankurk99) [inline src/plugin/kubearmor.go]: this won't be required as we are already checking for nil pointer above: line 389

File: src/plugin/kubearmor.go [hunk 2]
```diff
@@ -496,7 +495,9 @@ func StartKubeArmorRelay(StopChan chan struct{}, cfg types.ConfigKubeArmorRelay)
 		defer func() {
 			log.Info().Msg("watchalerts returning")
 			KubeArmorRelayStarted = false
-			_ = conn.Close()
+			if conn != nil {
+				_ = conn.Close()
+			}
```
Comment (Ankurk99) [inline src/plugin/kubearmor.go]: It's also not required

File: src/systempolicy/systemPolicy.go [hunk 3]
```diff
@@ -1468,31 +1468,33 @@ func DiscoverSystemPolicyMain() {
...
-		if cfg.GetCfgSystemLogFrom() == "kubearmor" {
-			url := cluster.GetKubearmorRelayURL()
-			if url == "" {
-				log.Error().Msg("kubearmor-relay url not found, retrying...")
-				for i := 0; i < types.Maxtries; i++ {
-					time.Sleep(10 * time.Second)
-					url = cluster.GetKubearmorRelayURL()
-					if url != "" {
-						break
+		if !plugin.KubeArmorRelayStarted {
```
Comment (Ankurk99) [inline src/systempolicy/systemPolicy.go]: > if !plugin.KubeArmorRelayStarted 


should be outside `for` loop

**Review by Ankurk99 — CHANGES_REQUESTED** [review]
some suggestions inline

**Comment by seswarrajan** [comment]
Closing and recreating new PR

**Pull request closed without merging**